### PR TITLE
CHA-350: CatalogsForApps - Add missing endpoints (variants & models)

### PR DESCRIPTION
**Pull request by hogren**
The goal of this PR is to update the API Ref with new endpoints for get mapped models and variants. I adapt some things like the link which redirect to the GET endpoints, the description of the already existing endpoint to tell about simple product OR flatten products.

A quick overview of the new titles and descriptions : 
image
image
image
image

I also create examples for the two new endpoints, don't hesitate to take a look.

### CODE REVIEW

**Comment by akeneo-secops** [comment]
### Gitleaks PR check
🟢 We detected no secrets within this PR! Good job 🎉

**Review by monikaakeneo — APPROVED** [review]

File: content/swagger/resources/app_catalogs/routes/app_catalogs_id_mapped_products.yaml [hunk 1]
```diff
@@ -1,13 +1,16 @@
 get:
-  summary: Get the list of mapped products related to a catalog
+  summary: Get the list of mapped fattened products or simple products related to a catalog
```
Comment (AdelELHAIBA-Akeneo) [inline content/swagger/resources/app_catalogs/routes/app_catalogs_id_mapped_products.yaml:2]:
```suggestion
  summary: Get the list of mapped flattened products or simple products related to a catalog
```

Comment (AdelELHAIBA-Akeneo) [reply content/swagger/resources/app_catalogs/routes/app_catalogs_id_mapped_products.yaml:2]: On retourne tjs des flattened products, ou product that correspond to the catalog mapping ?

File: content/swagger/resources/app_catalogs/routes/app_catalogs_id_mapped_products.yaml [hunk 2]
```diff
@@ -1,13 +1,16 @@
...
   tags:
     - Catalog products
   x-versions:
     - "SaaS"
-  description: This endpoint allows you to get the list of products related to a catalog when the mapping is enabled. 
-    Please, note that a disabled catalog can return an HTTP 200 with a payload containing an error message,
-    for more details see the <a href="apps/catalogs.html#troubleshooting">App Catalog</a> section.
+  description: This endpoint is available when the mapping is enabled on a specified catalog. When the mapping is in a
+    basic mode, this endpoint allows you to get the list of all fattened products related to the catalog. When it's in
+    models/variants mode, this endpoint allows you to get the list of simple products with all specified targets in the
```
Comment (AdelELHAIBA-Akeneo) [inline content/swagger/resources/app_catalogs/routes/app_catalogs_id_mapped_products.yaml:10]: On a vraiment le two level mapping ? je pensais qu'on l'avait supprimé suite au bug remonté par OSF

Comment (AdelELHAIBA-Akeneo) [reply content/swagger/resources/app_catalogs/routes/app_catalogs_id_mapped_products.yaml:10]: Si c'est pas supprimé, ne devrait-on pas le faire ASAP et ne pas le documenter ?

**Comment by clovis-akeneo** [comment]
Petite correction tant qu'on y est :pray: 
image
La description du _path parameters_ `id` est différente sur cet endpoint des autres, il faudrait mettre `Catalog ID` comme le reste

**Review by guillaumefer-akeneo — APPROVED** [review]

**Pull request merged**